routes/user.js

```diff
@@ -76,28 +82,260 @@ router.post('/library', function(req, res, next) {
+                        msg: 'deleting done'
+                    })
+                }
+            })
+        })
+        // ExportSong.update({_id: playlistID}, {
+        //     $pull: {
+        //         translations: translationID
+        //     }
+        // }, {
+        //     multi: true
+        // }, function(err) {
+        //     if (err) return handleError(err)
+        //     res.send({
+        //         msg: 'deleting done'
+        //     })
+        // })
+})
+
+router.get('/playlist/:playlist_name/export2', function(req, res) {
+    // var playlistName = req.params.playlist_name
+    // Playlist.find({owner: req.user._id, name: playlistName})
+    // // .populate(songs)
+    res.render('export2', {
+        playlistName: req.params.playlist_name
+    })
+})
+
+router.get('/playlist/:playlist_name/export3', function(req, res) {
+    var playlistName = req.params.playlist_name
+    var songss = []
+    var temp = []
+    Playlist.findOne({
+        owner: req.user._id,
+        name: playlistName
+    }, function(err, playlist) {
+        if (err) return handleError(err)
+        ExportSong.find({
+                owner: playlist._id
+            })
+            .populate('song translations')
+            .exec(function(err, songs) {
+                // console.log(songs)
+                if (err) return handleError(err)
+                songs.forEach((s, i, arr) => {
+                    temp = []
+                    temp.push(s.song)
+                    s.translations.forEach((t) => {
+                            temp.push(t)
+                        })
+                        // console.log(temp)
+                    songss.push(temp)
+                    if (i === arr.length - 1) {
+                        console.log(songss)
+                        res.render('export3', {
+                            songss: songss
+                        })
+                    }
+                })
+            })
+    })
+
+})
+
+router.post('/playlist/:playlist_name/export3', function(req, res) {
+    var playlistName = req.params.playlist_name
+    var songss = []
+    var temp = []
+    var filename = Date.now()
+    Playlist.findOne({
+        owner: req.user._id,
+        name: playlistName
+    }, function(err, playlist) {
+        if (err) return handleError(err)
+        ExportSong.find({
+                owner: playlist._id
+            })
+            .populate('song translations')
+            .exec(function(err, songs) {
+                // console.log(songs)
+                if (err) return handleError(err)
+                songs.forEach((s, i, arr) => {
+                    temp = []
+                    temp.push(s.song)
+                    s.translations.forEach((t, i) => {
+                        temp.push(t)
+                    })
+                    s.translations = []
+                    s.save()
+                    songss.push(temp)
+                    if (i === arr.length - 1) {
```

Comment:
is there a reason why you do this here?
What if you just let the foreach complete? Then you could do an app.render